models/articles.model.js

```diff
@@ -0,0 +1,21 @@
+const db = require("../db/connection");
+
+exports.selectAllArticles = () => {
+  return db
+    .query(
+      "SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comment_id) AS comment_count FROM articles JOIN comments ON comments.article_id = articles.article_id GROUP BY articles.article_id ORDER BY articles.created_at DESC;"
+    )
+    .then(({ rows }) => {
+      if (!rows.length) {
+        return Promise.reject({
+          status: 404,
+          message: "Not Found",
+        });
+      }
+
+      return rows.map((row) => {
+        row.comment_count = Number(row.comment_count);
```

Comment:
I don't think this map is necessary - have a look at what rows looks like upon its return